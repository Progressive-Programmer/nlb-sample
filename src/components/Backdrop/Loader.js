import React from 'react';
import PropTypes from 'prop-types';
import Backdrop from '@mui/material/Backdrop';
import CircularProgress from '@mui/material/CircularProgress';

const Loader = (props) => {
  const { isOpen } = props;

  return (
    <Backdrop sx={{ color: '#fff', zIndex: (theme) => theme.zIndex.drawer + 1 }} open={isOpen}>
      <CircularProgress color="inherit" />
    </Backdrop>
  );
};

export default Loader;

Loader.propTypes = {
  isOpen: PropTypes.bool
};

Loader.defaultProps = {
  isOpen: false
};
