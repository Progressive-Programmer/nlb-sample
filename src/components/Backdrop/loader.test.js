import React from 'react';
import { shallow } from 'enzyme';
import Backdrop from '@mui/material/Backdrop';
import CircularProgress from '@mui/material/CircularProgress';
import Loader from './Loader';

const setUp = (props = {}) => {
  const wrapper = shallow(<Loader {...props} />);
  return wrapper;
};

describe('Loader', () => {
  describe('has props', () => {
    let wrapper;
    let props;
    beforeEach(() => {
      props = {
        isOpen: false
      };
      wrapper = setUp(props);
    });
    afterEach(() => {
      jest.resetAllMocks();
    });

    it('should render correctly', () => {
      const component = wrapper;
      expect(component).toMatchSnapshot();
    });
    it('should not render Backdrop', () => {
      const component = wrapper;
      expect(component.contains(<Backdrop open={props.isOpen} />)).toEqual(false);
    });
    it('should render Backdrop', () => {
      const component = shallow(<Loader isOpen={true} />);
      expect(component.find(Backdrop)).toHaveLength(1);
      expect(component.contains(<CircularProgress color="inherit" />)).toEqual(true);
    });
  });
});
