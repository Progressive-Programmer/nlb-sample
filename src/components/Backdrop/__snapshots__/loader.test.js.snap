// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader has props should render correctly 1`] = `
<ForwardRef(Backdrop)
  open={false}
  sx={
    Object {
      "color": "#fff",
      "zIndex": [Function],
    }
  }
>
  <ForwardRef(CircularProgress)
    color="inherit"
  />
</ForwardRef(Backdrop)>
`;
